src/rust/demikernel/libos/mod.rs

```diff
@@ -159,22 +160,70 @@ impl LibOS {
 
     /// Waits for a pending operation in an I/O queue.
     pub fn wait(&mut self, qt: QToken) -> Result<demi_qresult_t, Fail> {
-        match self {
-            LibOS::NetworkLibOS(libos) => libos.wait(qt),
+        trace!("wait(): qt={:?}", qt);
+
+        // Retrieve associated schedule handle.
+        let handle: SchedulerHandle = self.schedule(qt)?;
+
+        loop {
+            // Poll first, so as to give pending operations a chance to complete.
+            self.poll();
+
+            // The operation has completed, so extract the result and return.
+            if handle.has_completed() {
+                return Ok(self.pack_result(handle, qt)?);
+            }
         }
     }
 
     /// Waits for an I/O operation to complete or a timeout to expire.
     pub fn timedwait(&mut self, qt: QToken, abstime: Option<SystemTime>) -> Result<demi_qresult_t, Fail> {
```

Comment:
Just out of curiosity, do you know what this variable name is supposed to mean?  And is it pronounced "A.B.S. time" or "abs time", or ...?  From its use, I would have thought it would be named "timeout" or "expiration".

Reply:
Maybe it is short for "absolute timeout" (as opposed to a relative timeout).

Reply:
@BrianZill when designing the `timedwait()` system call, I sought some inspiration in POSIX [`sem_timedwait()`](https://pubs.opengroup.org/onlinepubs/9699919799/nframe.html), since we did similarly for other calls.

In my understanding, this is an absolute timeout, because the standard has the following sentence:

"The timeout shall expire when the absolute time specified by abstime passes"

What do you think?